verilog/analysis/verilog_project.cc

```diff
@@ -185,6 +186,34 @@ bool VerilogProject::RemoveRegisteredFile(
   return false;
 }
 
+std::string VerilogProject::GetRelativePathToSource(
+    const absl::string_view absolute_filepath) {
+  // TODO add check if the absolute_filepath is out of the VerilogProject
+  std::filesystem::path absolutepath{absolute_filepath.begin(),
+                                     absolute_filepath.end()};
+  auto root = TranslationUnitRoot();
+  std::filesystem::path projectpath{root.begin(), root.end()};
+  auto relative = std::filesystem::relative(absolutepath, projectpath);
+  return relative.string();
+}
+
+absl::Status VerilogProject::UpdateFileContents(
+    absl::string_view path, const verible::TextStructureView* updatedtext) {
+  std::string projectpath = GetRelativePathToSource(path);
+  auto fileptr = files_.find(projectpath);
+  if (fileptr == files_.end()) {
+    files_.insert(
+        std::make_pair(projectpath, std::make_unique<ParsedVerilogSourceFile>(
+                                        projectpath, path, updatedtext,
+                                        /*corpus=*/"")));
+  } else {
+    fileptr->second = std::make_unique<ParsedVerilogSourceFile>(
+        projectpath, path, updatedtext,
+        /*corpus=*/"");
+  }
+  return absl::OkStatus();
```

Comment:
Currently, the only outcome of this function is `absl::OkStatus()`.
So until we actually have a status to return, let's just return `void` (simplifies all the places that call it).